src/os/bluestore/BlueFS.cc

```diff
@@ -831,8 +831,10 @@ void BlueFS::_stop_alloc()
 {
   dout(20) << __func__ << dendl;
   for (auto p : bdev) {
-    if (p)
+    if (p) {
+      p->set_discard_thread_notify();
```

Comment:
Looks like one can set the flag inside discard_drain instead of prepending discard_drain() call with this flag.

Reply:
And please do not forget to unset it on completion.

Reply:
done